core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -681,6 +694,17 @@ class GroupCoordinator(val brokerId: Int,
   def onCompleteJoin(group: GroupMetadata) {
     var delayedStore: Option[DelayedStore] = None
     group synchronized {
+      if (group.generationId == 0) {
+        if (group.newMemberAdded && time.milliseconds() < group.initialRebalanceTimeout) {
+          group.newMemberAdded = false
+          val delay = scala.math.min(groupConfig.groupInitialRebalanceDelayMs, group.initialRebalanceTimeout - time.milliseconds()).toInt
+          val groupKey = GroupKey(group.groupId)
+          joinPurgatory.watchOperation(new DelayedJoin(this, group, delay, group.initialRebalanceTimeout), Seq(groupKey))
```

Comment:
EDIT: now I realized that this is not triggered because we only call `checkAndComplete` as mentioned above:

https://github.com/apache/kafka/pull/2758#discussion_r113572998

so I think your current impl is actually better than my comments. NVM.